conbench/types.py

```diff
@@ -0,0 +1,11 @@
+# Do this for case ID, etc, too.
+
+from typing import NewType, Tuple
+
+TBenchmarkName = NewType("TBenchmarkName", str)
+
+
+# This type is used often. It's the famous 4-tuple defining a time series:
+# benchmark name, hardware ID, context ID, case ID. Maybe turn this into a
+# namedtuple or sth like this. Watch out a bit for mem consumption.
+Tt4 = Tuple[TBenchmarkName, str, str, str]
```

Comment:
This name is inscrutable to me. Maybe we need a better-defined name for the concept itself.

Reply:
> This name is inscrutable to me

Yes. Slowly converging to something that's hopefully useful. 

This type (well, the concept) is going to be so important, touching so many parts of the code base. 

It will pop up in other type definitions quite a bit so I like to have a short, recognizable term. 

Example:
```
TDict4tdf = Dict[Tt4, pd.DataFrame]
TDict4tlist = Dict[Tt4, List[BMRTBenchmarkResult]]
```

Totally work in progress.

https://github.com/conbench/conbench/blob/edeaf66832ca2eaa8446d79c3a8dac6b3c3d368d/conbench/job.py#L118

> It's the famous 4-tuple

Maybe in the future this is going to be a 5-tuple:

```
Tuple[TRepoUrl, TBenchmarkName, TCaseID, TContextID, THardwareHash]
```

Again, still various rough edges but we can start passing timeseries objects around that are self-documenting and hopefully easy to work with, sth in the lines of

```
class ResultTimeseries:
    t4: Tt4
    benchmark_name: str
    df: pd.DataFrame
    bmrlist: List[BMRTBenchmarkResult]
```

---

here, in this PR I just moved the definition of `Tt4` out of job.py because it starts to be re-used (as anticipated).